lib/indexer.rb

```diff
@@ -0,0 +1,63 @@
+module Argo
+  class Indexer
+    ##
+    # Returns a Logger instance for recording info about indexing attempts
+    # @yield attempt to execute 'entry_id_block' and use the result as an extra identifier for the log 
+    #   entry.  a placeholder will be used otherwise. 'request.uuid' might be useful in a Rails app.
+    def self.generate_index_logger(&entry_id_block)
+      index_logger = Logger.new(Settings.INDEXER.LOG, Settings.INDEXER.LOG_ROTATION_INTERVAL)
+      index_logger.formatter = proc do |severity, datetime, progname, msg|
+        date_format_str = Argo::Config.date_format_str
+        entry_id = begin entry_id_block.call rescue '---' end
```

Comment:
i get the sense that the advice there is about not being overly broad when catching exceptions, which i think is a very good rule of thumb.  but in this case, i think we want to trap anything that's `StandardError` and proceed with logging, since that seems more useful than dying.  for instance, even if the caller gave a block that worked incorrectly and always failed, i think it's better to otherwise log what that caller wants if possible, than it is to fail out of the logging call.

more catastrophic things should be `Exception`s, which aren't trapped by the default `rescue`, so those should still bubble up, which also seems right to me.